v2/internal/parse_test.go

```diff
@@ -28,23 +33,180 @@ func TestParseMessageFileBytes(t *testing.T) {
 				}},
 			},
 		},
+		{
+			name: "basic test with dot separator in key",
+			file: `{"prepended.hello": "world"}`,
+			path: "en.json",
+			messageFile: &MessageFile{
+				Path:   "en.json",
+				Tag:    language.English,
+				Format: "json",
+				Messages: []*Message{{
+					ID:    "prepended.hello",
+					Other: "world",
+				}},
+			},
+		},
+		{
+			name: "invalid test (no key)",
+			file: `"hello"`,
+			path: "en.json",
+			err:  errors.New("invalid translation file, expected key-values, got a single value"),
```

Comment:
Done, commit `87b9968`

Reply:
I was actually wondering about saving the error itself instead of just the message because I think that would simplify the code in the tests that checks for the equality of errors.

e.g.

```go

var errInvalidTranslationFile = errors.New("invalid translation file, expected key-values, got a single value")
```

Reply:
no problem, replaced commit 87b9968 with https://github.com/nicksnyder/go-i18n/commit/6d4c4423af15735809ce0e755a158228b56f4d21